### PR TITLE
Add storage secrets to es-index-cleaner cronjob

### PR DESCRIPTION
**Pull request by gregoryfranklin**
Resolves https://github.com/jaegertracing/jaeger-operator/issues/196

Related to https://github.com/jaegertracing/jaeger/pull/1318 which has been merged and released as `jaegertracing/jaeger-es-index-cleaner:latest`.  We're just adding some additional environment variables so its also fine to use with older versions of the jaeger-es-index-cleaner image.

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/197)

File: pkg/cronjob/es_index_cleaner.go [hunk 1]
```diff
@@ -17,6 +17,18 @@ func CreateEsIndexCleaner(jaeger *v1alpha1.Jaeger) *batchv1beta1.CronJob {
 	esUrls := getEsHostname(jaeger.Spec.Storage.Options.Map())
 	trueVar := true
 	name := fmt.Sprintf("%s-es-index-cleaner", jaeger.Name)
+
+	var envFromSource []v1.EnvFromSource
+	if len(jaeger.Spec.Storage.SecretName) > 0 {
+		envFromSource = append(envFromSource, v1.EnvFromSource{
+			SecretRef: &v1.SecretEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: jaeger.Spec.Storage.SecretName,
```
Comment (pavolloffay) [inline pkg/cronjob/es_index_cleaner.go]: @jpkrohling I am a bit outdatted with represents secretName in the storage spec?

Comment (annanay25) [reply pkg/cronjob/es_index_cleaner.go]: @pavolloffay I'm guessing its https://github.com/jaegertracing/jaeger-operator#secrets-support

**Comment by jpkrohling** [comment]
@pavolloffay , @gregoryfranklin , what's the status of this PR?

**Comment by pavolloffay** [comment]
Adding a yaml with a secret I have used.

```yaml
apiVersion: v1
kind: Secret
metadata:
  name: database-secrets
type: Opaque
data:
  ES_USERNAME: username
  ES_PASSWORD: password
```

**Comment by pavolloffay** [comment]
thanks @gregoryfranklin

**Pull request merged**